internal/node/node_repositories.bzl

```diff
@@ -198,13 +379,40 @@ def node_repositories(package_json):
 
   Args:
     package_json: a list of labels, which indicate the package.json files that need to be installed.
+    node_version: optional; the specific version of NodeJS to install.
+    yarn_version: optional; the specific version of Yarn to install.
   """
   # Windows users need sh_binary wrapped as an .exe
   check_bazel_version("0.5.4")
 
-  _node_repo(name = "nodejs", package_json = package_json)
+  # If the caller already executed node_download_runtime or node_local_runtime, and correctly used the "nodejs"
+  # repo name, then we can skip this step.
+  if "nodejs" not in native.existing_rules():
+    if node_version in NODE_REPOSITORIES:
+      node_download_runtime(
+          name = "nodejs",
+          version = node_version,
+          packages = NODE_REPOSITORIES[node_version],
+          package_json = package_json,
+      )
+    else:
+      fail("Unknown NodeJS version {}".format(node_version))
 
-  _yarn_repo(name = "yarn", package_json = package_json)
+  # If the caller already executed yarn_download or yarn_local, and correctly used the "yarn" repo name, then we can
+  # skip this step.
+  if "yarn" not in native.existing_rules():
+    if yarn_version in YARN_REPOSITORIES:
+      filename, strip_prefix, sha256 = YARN_REPOSITORIES[yarn_version]
+      yarn_download(
+        name = "yarn",
+        version = yarn_version,
+        filename = filename,
+        strip_prefix = strip_prefix,
+        sha256 = sha256,
+        package_json = package_json,
+      )
+    else:
+      fail("Unknown Yarn version {}".format(yarn_version))
```

Comment:
how would the user know which versions are supported?

Reply:
We could try to keep this list in sync with something in the top level readme?  I'll add something there and we can see how it works out.